yang-module-versioning/draft-ietf-netmod-yang-module-versioning.xml

```diff
@@ -772,36 +764,48 @@ module ietf-yang-revisions {
   // RFC Ed.: replace XXXX (inc above) with actual RFC number and
   // remove this note.
 
-  revision 2022-10-10 {
+  revision 2022-11-21 {
     rev:revision-label 1.0.0-draft-ietf-netmod-yang-module-versioning-07;
     description
       "Initial version.";
     reference
       "XXXX: Updated YANG Module Revision Handling";
   }
 
+  typedef revision-date {
+    type string {
+      pattern '[0-9]{4}-(1[0-2]|0[1-9])-(0[1-9]|[1-2][0-9]|3[0-1])';
+    }
+    description
+      "A date associated with a YANG revision.
+
+      Matches dates formatted as YYYY-MM-DD.";
+    reference
+      "RFC 7950: The YANG 1.1 Data Modeling Language";
+  }
+
   typedef revision-label {
     type string {
       length "1..255";
       pattern '[a-zA-Z0-9,\-_.+]+';
-      pattern '\d{4}-\d{2}-\d{2}' {
+      pattern '[0-9]{4}-(1[0-2]|0[1-9])-(0[1-9]|[1-2][0-9]|3[0-1])' {
+        error-message "The revision-label must not match a revision-date.";
         modifier invert-match;
       }
     }
     description
       "A label associated with a YANG revision.
 
-      Alphanumeric characters, comma, hyphen, underscore, period 
-      and plus are the only accepted characters. MUST NOT match 
```

Comment:
Has "MUST NOT match revision-date" been removed from the description because of the invert-match above?

Reply:
> Has "MUST NOT match revision-date" been removed from the description because of the invert-match above?

That's what I assumed and I think it is probably fine. We talked on a recent call about being OK with reducing duplication.

Reply:
FWIW, I was on the edge of keeping it in the description, even though
an error-message in the pattern was added.

Reply:
I don't feel strongly about this. I would have kept it.